### DISCUSSION TITLE
oAuth2.0 in this library (client id and client secret)?

### DISCUSSION THREAD

**Discussion by JPedroGon** [slack-go/slack#984] [General]
Hi,

I was trying to find if this lib already support the new way of authentication oAuth2 (client id and client secret) instead of the deprecated token ?

I could not find in the basic examples or in the api a reference but I might have searched wrongly. Anyone can enlightenment in this one or give a quick example of code?

Appreciated!

**Comment by jabalsad** [comment]
It's undocumented, but the source is here:
https://github.com/slack-go/slack/blob/9cad061143f5520f12dc1c707082fe8a6ab148c9/oauth.go
